src/main/java/io/quarkus/gizmo/MethodCheckerVisitor.java

```diff
@@ -0,0 +1,220 @@
+package io.quarkus.gizmo;
+
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.Attribute;
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.TypePath;
+
+public class MethodCheckerVisitor extends MethodVisitor {
+    private boolean isReturningValue;
+    private boolean isSuperOrThisCalled;
+
+    public MethodCheckerVisitor() {
+        this(Gizmo.ASM_API_VERSION);
+    }
+    public MethodCheckerVisitor(int api) {
+        super(api);
+        isReturningValue = false;
+        isSuperOrThisCalled = false;
+    }
+
+
+    public void check(MethodCreatorImpl methodCreator) {
+        visitLabel(methodCreator.getTop());
+        for (BytecodeCreatorImpl.Operation op : methodCreator.operations) {
+            op.writeBytecode(this);
+        }
+        visitLabel(methodCreator.getBottom());
+    }
+
+    @Override
+    public void visitInsn(int opcode) {
+        // if returnValue or ThrowException it is valid method.
+        switch (opcode) {
+            case Opcodes.RETURN:
+            case Opcodes.IRETURN:
+            case Opcodes.LRETURN:
+            case Opcodes.FRETURN:
+            case Opcodes.DRETURN:
+            case Opcodes.ARETURN:
+            case Opcodes.ATHROW:
+                isReturningValue = true;
+                break;
+        }
+    }
+
+    @Override
+    public void visitParameter(String name, int access) {
+    }
+
+    @Override
+    public AnnotationVisitor visitAnnotationDefault() {
+        return new DummyVisitor();
```

Comment:
why not return null like https://github.com/llbit/ow2-asm/blob/master/src/org/objectweb/asm/MethodVisitor.java ?

Reply:
current code do not handle it :
AnnotationUtils.visitAnnotationValue do not check if first param is null.
But you are right asm standard said that it can return null if no visitor is needed.
Other solution is to return null and check it in
https://github.com/quarkusio/gizmo/blob/bda4d0794f8b808c095c1351ba6bc48f92f50403/src/main/java/io/quarkus/gizmo/AnnotationUtils.java#L9

Reply:
@gsmet @stuartwdouglas  I provide 2 solutions for that. current code with dummy visitor or return null and handle it in AnnotationUtils. What do you prefer ?
Anyway, I need a full review before going further.